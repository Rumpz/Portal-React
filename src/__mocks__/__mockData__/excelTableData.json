{"headers":["MERGE2","TIPO_SINTOMA2","AREA_SINTOMA2","SUB_AREA_SINTOMA2","Incidência do Sintoma2","Obs2","OBSXX2","OBSvvv4"],"data":[["APPS TV IRIS | APP COM ERRO | MODEM NÃO REGISTA","APPS TV IRIS","APP COM ERRO","MODEM NÃO REGISTA","TV","-","-","-"],["APPS TV IRIS | ERRO NO ACESSO | DIAGNÓSTICO INCONCLUSIVO","APPS TV IRIS","ERRO NO ACESSO","DIAGNÓSTICO INCONCLUSIVO","TV","-","-","-"],["APPS TV IRIS | INFORMAÇÃO INDISPONÍVEL | DIAGNÓSTICO INCONCLUSIVO","APPS TV IRIS","INFORMAÇÃO INDISPONÍVEL","DIAGNÓSTICO INCONCLUSIVO","TV","-","-","-"],["APPS TV IRIS | ERRO NO ACESSO | PROBLEMA CONHECIDO","APPS TV IRIS","ERRO NO ACESSO","PROBLEMA CONHECIDO","TV","-","-","-"],["APPS TV IRIS | APP COM ERRO | MODEM NÃO REGISTA","APPS TV IRIS","APP COM ERRO","MODEM NÃO REGISTA","TV","-","-","-"],["APPS TV IRIS | ERRO NO ACESSO | DIAGNÓSTICO INCONCLUSIVO","APPS TV IRIS","ERRO NO ACESSO","DIAGNÓSTICO INCONCLUSIVO","TV","-","-","-"],["APPS TV IRIS | INFORMAÇÃO INDISPONÍVEL | DIAGNÓSTICO INCONCLUSIVO","APPS TV IRIS","INFORMAÇÃO INDISPONÍVEL","DIAGNÓSTICO INCONCLUSIVO","TV","-","-","-"],["APPS TV IRIS | ERRO NO ACESSO | PROBLEMA CONHECIDO","APPS TV IRIS","ERRO NO ACESSO","PROBLEMA CONHECIDO","TV","-","-","-"],["APPS TV IRIS | APP COM ERRO | MODEM NÃO REGISTA","APPS TV IRIS","APP COM ERRO","MODEM NÃO REGISTA","TV","-","-","-"],["APPS TV IRIS | ERRO NO ACESSO | DIAGNÓSTICO INCONCLUSIVO","APPS TV IRIS","ERRO NO ACESSO","DIAGNÓSTICO INCONCLUSIVO","TV","-","-","-"],["APPS TV IRIS | INFORMAÇÃO INDISPONÍVEL | DIAGNÓSTICO INCONCLUSIVO","APPS TV IRIS","INFORMAÇÃO INDISPONÍVEL","DIAGNÓSTICO INCONCLUSIVO","TV","-","-","-"],["APPS TV IRIS | ERRO NO ACESSO | PROBLEMA CONHECIDO","APPS TV IRIS","ERRO NO ACESSO","PROBLEMA CONHECIDO","TV","-","-","-"],["APPS TV IRIS | APP COM ERRO | MODEM NÃO REGISTA","APPS TV IRIS","APP COM ERRO","MODEM NÃO REGISTA","TV","-","-","-"],["APPS TV IRIS | ERRO NO ACESSO | DIAGNÓSTICO INCONCLUSIVO","APPS TV IRIS","ERRO NO ACESSO","DIAGNÓSTICO INCONCLUSIVO","TV","-","-","-"],["APPS TV IRIS | INFORMAÇÃO INDISPONÍVEL | DIAGNÓSTICO INCONCLUSIVO","APPS TV IRIS","INFORMAÇÃO INDISPONÍVEL","DIAGNÓSTICO INCONCLUSIVO","TV","-","-","-"],["APPS TV IRIS | ERRO NO ACESSO | PROBLEMA CONHECIDO","APPS TV IRIS","ERRO NO ACESSO","PROBLEMA CONHECIDO","TV","-","-","-"]]}