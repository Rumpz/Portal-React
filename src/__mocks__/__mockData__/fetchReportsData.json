[
  {
    "id":1,
    "codigo_coisas":"1-CVERTTV",
    "outros_codigos":"Done"
  },
  {
    "id":2,
    "codigo_coisas":"1-XXXXXXX",
    "outros_codigos":"Almost Done"
  },
  {
    "id":3,
    "codigo_coisas":"1-YYYYYYY",
    "outros_codigos":"Doing it"
  }
]