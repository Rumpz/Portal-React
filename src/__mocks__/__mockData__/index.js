module.exports = {
  fetchReportsData: require('./fetchReportsData'),
  excelTableData: require('./excelTableData')
};
