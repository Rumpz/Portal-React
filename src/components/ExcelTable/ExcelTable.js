import React from 'react';
import './css/excelTable.css';

const ExcelTable = (props) => {
  return (
    <div>
      <h4>Excel Table Viewer</h4>
      <table>
        <thead>
          <tr key={'headers'}>{props.cols.map((c, i) =>
            <th key={i}>{c.name}</th>)}
          </tr>
        </thead>
        <tbody>
          {props.data.map((r, i) => <tr key={r[0] + i}>
            {props.cols.map(c => <td key={c.key + i}>{ r[c.key]}</td>)}
          </tr>)}
        </tbody>
      </table>
    </div>
  );
};

export default ExcelTable;
