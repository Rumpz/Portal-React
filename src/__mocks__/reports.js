/* // __mocks__/request.js
import { fetchReportsData } from './__mockData__';

export default function fetchReports(url) {
  return new Promise((resolve, reject) => {
    process.nextTick(
      () =>
        url === 'reporting'
          ? resolve(fetchReportsData)
          : reject({
              error: 'Status 404 Not Found',
            }),
    );
  });
} */

export default {
  get: jest.fn(() => Promise.resolve({ data: {} }))
};