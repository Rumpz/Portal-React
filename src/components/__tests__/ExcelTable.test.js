import EnzymeAdapter from 'enzyme-adapter-react-16';
import ShallowRenderer from 'react-test-renderer/shallow';
import Enzyme from 'enzyme';
import React from 'react';
import ExcelTable from '../ExcelTable/ExcelTable';
import { excelTableData } from '../../__mocks__/__mockData__';
// This sets up the adapter to be used by Enzyme
Enzyme.configure({ adapter: new EnzymeAdapter() });

// Test Reports Component Render
// It should render an unordered list with the given data drray
it('renders reports list correctly', async () => {
  const renderer = new ShallowRenderer();
  renderer.render(<ExcelTable data={excelTableData['headers']} cols={excelTableData['data']} />);
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});
