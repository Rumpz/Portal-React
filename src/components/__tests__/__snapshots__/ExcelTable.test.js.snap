// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders reports list correctly 1`] = `
<div>
  <h4>
    Excel Table Viewer
  </h4>
  <table>
    <thead>
      <tr>
        <th />
        <th />
        <th />
        <th />
        <th />
        <th />
        <th />
        <th />
        <th />
        <th />
        <th />
        <th />
        <th />
        <th />
        <th />
        <th />
      </tr>
    </thead>
    <tbody>
      <tr>
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
      </tr>
      <tr>
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
      </tr>
      <tr>
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
      </tr>
      <tr>
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
      </tr>
      <tr>
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
      </tr>
      <tr>
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
      </tr>
      <tr>
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
      </tr>
      <tr>
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;
